Panic when receiving same pubsub message

To reproduce, use index-provider, import car file, and then use `provider announce` multiple times in a short amount of time.  This will cause the indexer to panic with the following:
```
panic: putting the same entry twice not supported                    
                                                                                                                                                 
goroutine 181 [running]:                                                                                                                         
github.com/libp2p/go-libp2p-pubsub/timecache.FirstSeenCache.Add({0x14001fdc9f0?, 0x14001fdca20?, 0x14001ff8000?, 0x14001f98048?}, {0x1400011e160, 0x20})
        /Users/ajg/go/pkg/mod/github.com/libp2p/go-libp2p-pubsub@v0.9.0/timecache/first_seen_cache.go:32 +0x264
github.com/libp2p/go-libp2p-pubsub.(*PubSub).markSeen(0x14001ffc000, {0x1400011e160, 0x20})
        /Users/ajg/go/pkg/mod/github.com/libp2p/go-libp2p-pubsub@v0.9.0/pubsub.go:1002 +0x124
github.com/libp2p/go-libp2p-pubsub.(*validation).validate(0x14001fac140, {0x0, 0x0, 0x0?}, {0x14001dc6210, 0x26}, 0x14001fac000, 0x0)
        /Users/ajg/go/pkg/mod/github.com/libp2p/go-libp2p-pubsub@v0.9.0/validation.go:288 +0xe8
github.com/libp2p/go-libp2p-pubsub.(*validation).validateWorker(0x14001fac140)
        /Users/ajg/go/pkg/mod/github.com/libp2p/go-libp2p-pubsub@v0.9.0/validation.go:266 +0x4c
created by github.com/libp2p/go-libp2p-pubsub.(*validation).Start
        /Users/ajg/go/pkg/mod/github.com/libp2p/go-libp2p-pubsub@v0.9.0/validation.go:133 +0x6c
```

--- a/go.mod
+++ b/go.mod
@@ -26,8 +26,8 @@ require (
 	github.com/ipld/go-ipld-prime v0.20.0
 	github.com/ipld/go-ipld-prime/storage/dsadapter v0.0.0-20230102063945-1a409dc236dd
 	github.com/ipld/go-storethehash v0.3.13
-	github.com/ipni/go-indexer-core v0.7.6-0.20230330202523-6ec70bd203fe
-	github.com/ipni/go-libipni v0.0.0-20230330175745-8950ad3901cc
+	github.com/ipni/go-indexer-core v0.7.5
+	github.com/ipni/go-libipni v0.0.2
 	github.com/libp2p/go-libp2p v0.26.4
 	github.com/libp2p/go-msgio v0.3.0
 	github.com/mitchellh/go-homedir v1.1.0

--- a/go.mod
+++ b/go.mod
@@ -87,7 +87,6 @@ require (
 	github.com/docker/go-connections v0.4.0 // indirect
 	github.com/docker/go-units v0.5.0 // indirect
 	github.com/elastic/gosigar v0.14.2 // indirect
-	github.com/emirpasic/gods v1.18.1 // indirect
 	github.com/filecoin-project/go-cbor-util v0.0.1 // indirect
 	github.com/filecoin-project/go-data-transfer/v2 v2.0.0-rc5 // indirect
 	github.com/filecoin-project/go-ds-versioning v0.1.2 // indirect
@@ -160,7 +159,7 @@ require (
 	github.com/libp2p/go-flow-metrics v0.1.0 // indirect
 	github.com/libp2p/go-libp2p-asn-util v0.2.0 // indirect
 	github.com/libp2p/go-libp2p-gostream v0.6.0 // indirect
-	github.com/libp2p/go-libp2p-pubsub v0.9.0 // indirect
+	github.com/libp2p/go-libp2p-pubsub v0.9.3 // indirect
 	github.com/libp2p/go-libp2p-record v0.2.0 // indirect
 	github.com/libp2p/go-nat v0.1.0 // indirect
 	github.com/libp2p/go-netroute v0.2.1 // indirect

--- a/go.mod
+++ b/go.mod
@@ -27,7 +27,7 @@ require (
 	github.com/ipld/go-ipld-prime/storage/dsadapter v0.0.0-20230102063945-1a409dc236dd
 	github.com/ipld/go-storethehash v0.3.13
 	github.com/ipni/go-indexer-core v0.7.5
-	github.com/ipni/go-libipni v0.0.3
+	github.com/ipni/go-libipni v0.0.4
 	github.com/libp2p/go-libp2p v0.26.4
 	github.com/libp2p/go-msgio v0.3.0
 	github.com/mitchellh/go-homedir v1.1.0

--- a/go.mod
+++ b/go.mod
@@ -26,7 +26,7 @@ require (
 	github.com/ipld/go-ipld-prime v0.20.0
 	github.com/ipld/go-ipld-prime/storage/dsadapter v0.0.0-20230102063945-1a409dc236dd
 	github.com/ipld/go-storethehash v0.3.13
-	github.com/ipni/go-indexer-core v0.7.5
+	github.com/ipni/go-indexer-core v0.7.6
 	github.com/ipni/go-libipni v0.0.4
 	github.com/libp2p/go-libp2p v0.26.4
 	github.com/libp2p/go-msgio v0.3.0

--- a/go.mod
+++ b/go.mod
@@ -27,7 +27,7 @@ require (
 	github.com/ipld/go-ipld-prime/storage/dsadapter v0.0.0-20230102063945-1a409dc236dd
 	github.com/ipld/go-storethehash v0.3.13
 	github.com/ipni/go-indexer-core v0.7.5
-	github.com/ipni/go-libipni v0.0.2
+	github.com/ipni/go-libipni v0.0.3
 	github.com/libp2p/go-libp2p v0.26.4
 	github.com/libp2p/go-msgio v0.3.0
 	github.com/mitchellh/go-homedir v1.1.0

